### PR TITLE
[RPA-956] Guest IP address anonymization

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Description

DONE:
1. Changes in GuestServiceImpl
2. Add new database changelog to change the guest column name to 'ip_hash'

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230919606868_changelog.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-4.1.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="depcikm" id="20230919606868-1">
+        <preConditions onFail="MARK_RAN">
+            <and>
+                <tableExists tableName="guest"/>
+                <not>
```
Comment (Aeri4a) [inline runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230919606868_changelog.xml:8]: Does it mean - do it when table exists AND column NOT exists? Don't we check if column actually exists to make sure that we can change its name?

File: runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230919606868_changelog.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+                    <columnExists tableName="guest" columnName="ip_hash"/>
+                </not>
+            </and>
+        </preConditions>
```
Comment (preludium) [inline runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20230919606868_changelog.xml]: include condition about ip column existance

**Review by preludium — APPROVED** [review]

File: runbotics-orchestrator/src/main/java/com/runbotics/service/impl/GuestServiceImpl.java [hunk 3]
```diff
@@ -39,7 +39,9 @@ public GuestServiceImpl(
     }
 
     public boolean verifyGuestLimit(String guestIp) {
-        Optional<Guest> guestOptional = guestRepository.findById(guestIp);
+        Optional<Guest> guestOptional = guestRepository.findAll().stream()
+            .filter(guest -> passwordEncoder.matches(guestIp, guest.getIp()))
+            .findFirst();
```
Comment (Aeri4a) [inline runbotics-orchestrator/src/main/java/com/runbotics/service/impl/GuestServiceImpl.java:44]: It looks good for me. I'm just wondering if there is another way to find just one guest user or it makes no difference?

Comment (marcin-depcik) [reply runbotics-orchestrator/src/main/java/com/runbotics/service/impl/GuestServiceImpl.java:44]: The reason why I did it this way is that bcrypt generates a unique hash every time, so we are not able to make such a query to the database that will check if the guest with that ip_hash already exists

**Pull request merged**